testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_predective_queries.py

```diff
@@ -0,0 +1,126 @@
+        doc_body = doc_generators.four_k()
+    elif doc_generator_type == "simple_user":
+        doc_body = doc_generators.simple_user()
+    elif doc_generator_type == "complex_doc":
+        doc_body = doc_generators.complex_doc()
+    else:
+        doc_body = doc_generators.simple()
+
+    db.create_bulk_docs(2, "cbl-predictive", db=cbl_db, generator=doc_generator_type)
+    result_set = predictive_query.getPredictionQueryResult(model, doc_body, cbl_db)
+
+    for result in result_set:
+        assert deep_dict_compare(doc_body, result[result.keys()[0]])
+
+    non_dict = "non_dict"
+    error = predictive_query.queryNonDictionaryInput(model, non_dict, cbl_db)
+    assert "Parameter of prediction() must be a dictionary" in error
+
+
+@pytest.mark.sanity
+@pytest.mark.listener
+@pytest.mark.predictivequeries
+@pytest.mark.parametrize("coordinates, euclidean_result, squareEuclidean_result, cosine_result",
+                         [
+                             ([[10, 10], [13, 14]], 5, 25, 0.0006851662332328923),
+                             ([[10, 0], [13, 14]], 14.317821063276, 205, 0.319548900632722),
+                             ([[1, 2, 3, 4], [12, 1]], None, None, None),
+                             ([[], [3]], None, None, None),
+                             ([None, [1, 3, 4]], None, None, None),
+                             ([[3, 6, 10], [34, 23, 4]], 35.86084215408221, 1286, 0.43620415177230565),
+                             ([[3, 6, 10, 34], [34, 23, 4, -45]], 86.75828490697589, 7527, 1.5677405830676259),
+                             ([[1, 2], "foo"], None, None, None)
+                         ]
+                         )
+def test_predictiveQueries_euclideanCosineDistance(params_from_base_test_setup, coordinates, euclidean_result,
+                                                   squareEuclidean_result, cosine_result):
+    '''
+        @summary:
+        1. Create docs with x,y,z coordinates
+        2. find the euclidean distance, square euclidean distance, and cosine distance
+        3. Verify all 3 distances return right value
+    '''
+    base_url = params_from_base_test_setup["base_url"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+
+    document = Document(base_url)
+
+    # Reset cluster to ensure no data in system
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    predictive_query = PredictiveQueries(base_url)
+
+    # Create dictionary and get the prediction query for euclidean distance
+    keys = ["abc", "def"]
+    doc_list = coordinates
+    mut_doc = document.create()
+    for item, key in zip(doc_list, keys):
+        document.setValue(mut_doc, key, item)
+    db.saveDocument(cbl_db, mut_doc)
+    result_euclidean_dist = predictive_query.euclideanDistance(cbl_db, keys[0], keys[1])
```

Comment:
add [0] to the end of this statement to avoid duplication assignment.

Reply:
done